libs/core-ui/src/lib/components/AxisConfigBinOptions.tsx

```diff
@@ -26,68 +28,120 @@ export interface IAxisConfigBinOptionsProps {
+          allowUserInteract(this.props.selectedColumn.property) && (
+            <Toggle
+              key="logarithmic-toggle"
+              label={localization.Interpret.AxisConfigDialog.logarithmicScaling}
+              inlineLabel
+              checked={selectedMeta.AxisType === AxisTypes.Logarithmic}
+              onChange={this.enableLogarithmicScaling}
             />
           )}
-        {!(this.props.mustBin || this.props.selectedColumn.options.bin) &&
-          this.props.canDither &&
+        {selectedMeta.featureRange?.rangeType === RangeTypes.Integer &&
           allowUserInteract(this.props.selectedColumn.property) && (
-            <Checkbox
-              key={this.props.selectedColumn.property}
-              label={localization.Interpret.AxisConfigDialog.ditherLabel}
-              checked={this.props.selectedColumn.options.dither}
-              onChange={this.ditherChecked}
+            <Toggle
+              key="categorical-toggle"
+              label={localization.Interpret.AxisConfigDialog.TreatAsCategorical}
+              inlineLabel
+              checked={selectedMeta.treatAsCategorical}
+              onChange={this.setAsCategorical}
             />
           )}
-      </>
+        {selectedMeta?.treatAsCategorical ? (
+          <>
+            <Text variant="small">
+              {selectedColumnDesc.categoricalDescription}
+            </Text>
+            {this.props.canDither && (
+              <Toggle
+                key="dither-toggle-categorical-checked"
+                label={localization.Interpret.AxisConfigDialog.ditherLabel}
+                inlineLabel
+                checked={this.props.selectedColumn.options.dither}
+                onChange={this.ditherChecked}
+              />
+            )}
+          </>
+        ) : (
+          <>
+            {this.props.canBin && !this.props.mustBin && (
+              <Toggle
+                key="bin-toggle"
+                label={localization.Interpret.AxisConfigDialog.binLabel}
+                inlineLabel
+                checked={this.props.selectedColumn.options.bin}
+                onChange={this.shouldBinClicked}
+              />
+            )}
+            {(this.props.mustBin || this.props.selectedColumn.options.bin) &&
+              this.props.selectedBinCount !== undefined && (
+                <AxisConfigDialogSpinButton
+                  label={localization.Interpret.AxisConfigDialog.numOfBins}
+                  max={this.props.maxHistCols}
+                  min={this.props.minHistCols}
+                  setNumericValue={this.setNumericValue}
+                  value={this.props.selectedBinCount.toString()}
+                />
+              )}
+            {!(this.props.mustBin || this.props.selectedColumn.options.bin) &&
+              this.props.canDither &&
+              allowUserInteract(this.props.selectedColumn.property) && (
+                <Toggle
```

Comment:
this Toggle looks the same, with same parameters, as the one in line 83:
https://github.com/microsoft/responsible-ai-toolbox/pull/1802/files#diff-635f8fa7b2f26365844135a68ebdb4ae8c323032fb332b0bf348c44d2154bbc8R83
consider refactoring both to a common method?

Reply:
Thanks for the comment! Updated